### PR TITLE
Update to CoreCLR FileStream

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Take the latest CoreCLR and use its copy of FileStream.

@ianhays @stephentoub @weshaggard @alexghi

### CODE REVIEW

**Comment by JeremyKuhne** [comment]
@stephentoub I put a SafeFileHandleHelper.Unix with the static method to open the FileStream. I'll make another pass to clean the helper out of SafeFileHandle when I undo direct references after the next CoreFX package update.

**Comment by stephentoub** [comment]
Is it possible to split this into multiple commits?  One containing the UpdateDependencies updates across project.json files, and another containing the manual changes you then did around FileStream, fixing conflicts, etc.?

**Comment by JeremyKuhne** [comment]
@stephentoub I'll look to see how one would do that.

**Comment by stephentoub** [comment]
> I'll look to see how one would do that.

Thanks.

>  I put a SafeFileHandleHelper.Unix with the static method to open the FileStream. I'll make another pass to clean the helper out of SafeFileHandle when I undo direct references after the next CoreFX package update.

I assume the code I see there is an exact copy of what it was when it was on SafeFileHandle?

**Comment by JeremyKuhne** [comment]
> I assume the code I see there is an exact copy of what it was when it was on SafeFileHandle?

Yes.

I figured out how to split it using VS- the .json files are all in the first commit. Everything else in the second.

**Comment by JeremyKuhne** [comment]
Updating for the new conflict in SRE...

**Comment by JeremyKuhne** [comment]
@dotnet-bot test Innerloop Linux ARM Emulator SoftFP Debug Cross Build (git clone timed out)

**Comment by JeremyKuhne** [comment]
Updating jsons again...

**Comment by JeremyKuhne** [comment]
After resolving up I'm getting this for everything:

```
CSC : error CS8034: Unable to load Analyzer assembly D:\Repos\corefx\Tools\net45\analyzers\System.Runtime.InteropServices.Analyzers.dll : Could not load file or assembly 'System.Runtime.InteropServices.An
alyzers, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. Strong name validation failed. (Exception from HRESULT: 0x8013141A) [D:\Repos\corefx\src\System.Net.
Mail\src\System.Net.Mail.csproj]
```

I don't see anything obvious- still investigating...

**Comment by JeremyKuhne** [comment]
No idea why the issue above is happening only locally for me. Disabled some tests that have differing results on Unix/Windows: #13247

File: src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoAbbreviatedDayNames.cs [hunk 1]
```diff
@@ -23,11 +23,11 @@ public void AbbreviatedDayNames_Set()
             Assert.Equal(newAbbreviatedDayNames, format.AbbreviatedDayNames);
         }
 
-        [Fact]
+        [Fact ActiveIssue(13247, TestPlatforms.AnyUnix)]
         public void AbbreviatedDayNames_Set_Invalid()
         {
             Assert.Throws<ArgumentNullException>("value", () => new DateTimeFormatInfo().AbbreviatedDayNames = null); // Value is null
-            Assert.Throws<ArgumentNullException>("value", () => new DateTimeFormatInfo().AbbreviatedDayNames = new string[] { "1", "2", "3", null, "5", "6", "7" }); // Value has null
+            Assert.Throws<ArgumentNullException>("values[3]", () => new DateTimeFormatInfo().AbbreviatedDayNames = new string[] { "1", "2", "3", null, "5", "6", "7" }); // Value has null
```
Comment (stephentoub) [inline src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoAbbreviatedDayNames.cs]: You're updating this value and disabling the test?  I suggest keeping the ActiveIssue but reverting this change, such that whoever investigates the behavior change can investigate whether it was intentional or not, and only if it was update the test to match.

File: src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoAbbreviatedMonthGenitiveNames.cs [hunk 2]
```diff
@@ -23,11 +23,11 @@ public void AbbreviatedMonthNames_Set()
             Assert.Equal(newAbbreviatedMonthGenitiveNames, format.AbbreviatedMonthGenitiveNames);
         }
 
-        [Fact]
+        [Fact ActiveIssue(13247, TestPlatforms.AnyUnix)]
         public void AbbreviatedMonths_Set_Invalid()
         {
             Assert.Throws<ArgumentNullException>("value", () => new DateTimeFormatInfo().AbbreviatedMonthGenitiveNames = null); // Value is null
-            Assert.Throws<ArgumentNullException>("value", () => new DateTimeFormatInfo().AbbreviatedMonthGenitiveNames = new string[] { "1", "2", "3", null, "5", "6", "7", "8", "9", "10", "11", "12", "" }); // Value has null
+            Assert.Throws<ArgumentNullException>("values[3]", () => new DateTimeFormatInfo().AbbreviatedMonthGenitiveNames = new string[] { "1", "2", "3", null, "5", "6", "7", "8", "9", "10", "11", "12", "" }); // Value has null
```
Comment (stephentoub) [inline src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoAbbreviatedMonthGenitiveNames.cs]: Ditto; same for all of these tests.

File: src/System.IO.FileSystem.Primitives/ref/System.IO.FileSystem.Primitives.cs [hunk 3]
```diff
@@ -5,7 +5,10 @@
 // Changes to this file must follow the http://aka.ms/api-review process.
 // ------------------------------------------------------------------------------
 
+[assembly: System.Runtime.CompilerServices.TypeForwardedTo(typeof(Microsoft.Win32.SafeHandles.SafeFileHandle))]
 [assembly: System.Runtime.CompilerServices.TypeForwardedTo(typeof(System.IO.FileAccess))]
+[assembly: System.Runtime.CompilerServices.TypeForwardedTo(typeof(System.IO.FileMode))]
+[assembly: System.Runtime.CompilerServices.TypeForwardedTo(typeof(System.IO.FileShare))]
 
 namespace System.IO
 {
```
Comment (stephentoub) [inline src/System.IO.FileSystem.Primitives/ref/System.IO.FileSystem.Primitives.cs]: We now have an entire contract assembly for a single enum?  Should FileAttributes be moved down as well, and then this contract can basically just be for compat?
cc: @weshaggard

File: src/System.IO.FileSystem/src/System/IO/UnixFileSystem.cs [hunk 4]
```diff
@@ -13,6 +13,8 @@ namespace System.IO
     /// <summary>Provides an implementation of FileSystem for Unix systems.</summary>
     internal sealed partial class UnixFileSystem : FileSystem
     {
+        internal const int DefaultBufferSize = 4096;
```
Comment (stephentoub) [inline src/System.IO.FileSystem/src/System/IO/UnixFileSystem.cs]: How about putting this const somewhere shared rather than duplicating it here and in File?

File: src/System.Net.Http/tests/UnitTests/System.Net.Http.Unit.Tests.csproj [hunk 5]
```diff
@@ -14,6 +14,8 @@
     <StringResourcesPath>../../src/Resources/Strings.resx</StringResourcesPath>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <NugetTargetMoniker Condition="'$(TargetGroup)' == ''">.NETStandard,Version=v1.7</NugetTargetMoniker>
+    <!-- HttpVersion is conflicting with System.Net.Primitives -->
+    <NoWarn>$(NoWarn);0436</NoWarn>
```
Comment (stephentoub) [inline src/System.Net.Http/tests/UnitTests/System.Net.Http.Unit.Tests.csproj]: This should no longer be necessary.

File: src/System.Runtime/ref/System.Runtime.cs [hunk 6]
```diff
@@ -13,24 +13,38 @@ public abstract partial class CriticalHandleMinusOneIsInvalid : System.Runtime.I
...
         public override bool IsInvalid { [System.Security.SecurityCriticalAttribute]get { throw null; } }
     }
+
     [System.Security.SecurityCriticalAttribute]
     public abstract partial class CriticalHandleZeroOrMinusOneIsInvalid : System.Runtime.InteropServices.CriticalHandle
     {
         protected CriticalHandleZeroOrMinusOneIsInvalid() : base(System.IntPtr.Zero) { }
         public override bool IsInvalid { [System.Security.SecurityCriticalAttribute]get { throw null; } }
     }
+
```
Comment (stephentoub) [inline src/System.Runtime/ref/System.Runtime.cs]: All of these blank lines will likely be removed the next time someone uses a tool to generate this file.  Probably best to just not add them.

File: src/System.Runtime/tests/System/Text/StringBuilderTests.cs [hunk 7]
```diff
@@ -1456,9 +1456,9 @@ public static void Remove_Invalid()
             Assert.Throws<ArgumentOutOfRangeException>("startIndex", () => builder.Remove(-1, 0)); // Start index < 0
             Assert.Throws<ArgumentOutOfRangeException>("length", () => builder.Remove(0, -1)); // Length < 0
 
-            Assert.Throws<ArgumentOutOfRangeException>("index", () => builder.Remove(6, 0)); // Start index + length > 0
-            Assert.Throws<ArgumentOutOfRangeException>("index", () => builder.Remove(5, 1)); // Start index + length > 0
-            Assert.Throws<ArgumentOutOfRangeException>("index", () => builder.Remove(4, 2)); // Start index + length > 0
+            Assert.Throws<ArgumentOutOfRangeException>("length", () => builder.Remove(6, 0)); // Start index + length > 0
+            Assert.Throws<ArgumentOutOfRangeException>("length", () => builder.Remove(5, 1)); // Start index + length > 0
+            Assert.Throws<ArgumentOutOfRangeException>("length", () => builder.Remove(4, 2)); // Start index + length > 0
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/Text/StringBuilderTests.cs]: It looks like these values changed in https://github.com/dotnet/coreclr/commit/a212082aa4cea8529b4c78844e2a7408ff6f7969.  @jkotas, @AlexGhiondea, @Clockwork-Muse, was this intentional?  If so, we'll need to specialize the test for coreclr vs desktop (or remove the name validation), as this change will cause the test to fail on desktop.

File: src/System.Text.Encoding/tests/Fallback/EncoderReplacementFallbackTests.cs [hunk 8]
```diff
@@ -109,7 +109,7 @@ public void CreateFallbackBuffer_Fallback_InvalidSurrogateChars_ThrowsArgumentOu
             EncoderFallbackBuffer buffer = new EncoderReplacementFallback().CreateFallbackBuffer();
 
             Assert.Throws<ArgumentOutOfRangeException>("charUnknownHigh", () => buffer.Fallback('a', '\uDC00', 0));
-            Assert.Throws<ArgumentOutOfRangeException>("CharUnknownLow", () => buffer.Fallback('\uD800', 'a', 0));
+            Assert.Throws<ArgumentOutOfRangeException>("charUnknownLow", () => buffer.Fallback('\uD800', 'a', 0));
```
Comment (stephentoub) [inline src/System.Text.Encoding/tests/Fallback/EncoderReplacementFallbackTests.cs]: Same comment here about the name change, cc: @tarekgh, @Clockwork-Muse

File: src/System.Runtime/ref/System.Runtime.cs [hunk 9]
```diff
@@ -4525,6 +4562,61 @@ public abstract partial class Stream : System.MarshalByRefObject, System.IDispos
...
+    public enum FileOptions
+    {
+        Asynchronous = 1073741824,
+        DeleteOnClose = 67108864,
+        Encrypted = 16384,
+        None = 0,
+        RandomAccess = 268435456,
+        SequentialScan = 134217728,
+        WriteThrough = -2147483648,
+    }
```
Comment (stephentoub) [inline src/System.Runtime/ref/System.Runtime.cs]: We're moving a bunch of types down to System.Runtime... we should move the corresponding tests as well.

File: src/System.Runtime/tests/System/Text/StringBuilderTests.cs (hunk 7 above)
Comment (jkotas) [inline src/System.Runtime/tests/System/Text/StringBuilderTests.cs]: >  If so, we'll need to specialize the test for coreclr vs desktop (or remove the name validation)

Is there a precedent for this?

Comment (jkotas) [inline src/System.Runtime/tests/System/Text/StringBuilderTests.cs]: Looking at the CoreCLR change:
- The coreclr change changed the name of `AddYears` argument name. It needs to reverted. Argument names are part of the contract.
- The name changes need to be ported to CoreRT so that we are consist

I think that the best course of action is to revert the offending part of the change in CoreCLR so that we can deal with it separately from the FileStream changes. I am going to do it.

Comment (stephentoub) [inline src/System.Runtime/tests/System/Text/StringBuilderTests.cs]: Sounds good.

Comment (jkotas) [inline src/System.Runtime/tests/System/Text/StringBuilderTests.cs]: I have done the rollbacks and creates:

https://github.com/dotnet/coreclr/issues/7952
https://github.com/dotnet/corert/issues/2124

cc @Clockwork-Muse

File: src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoAbbreviatedDayNames.cs (hunk 1 above)
Comment (tarekgh) [inline src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoAbbreviatedDayNames.cs]: I'll try to look at this later but this looks something weird to me as this used to work and I didn't touch this part recently. anyway we'll know when I investigate.

---
In reply to: [86105783](https://github.com/dotnet/corefx/pull/13229#discussion_r86105783) [](ancestors = 86105783)

File: src/System.IO.FileSystem/src/System/IO/Win32FileSystem.cs [hunk 10]
```diff
@@ -11,6 +11,8 @@ namespace System.IO
 {
     internal sealed partial class Win32FileSystem : FileSystem
     {
+        internal const int GENERIC_READ = unchecked((int)0x80000000);
```
Comment (tarekgh) [inline src/System.IO.FileSystem/src/System/IO/Win32FileSystem.cs]: >unchecked [](start = 42, length = 9)

is this really needed?

File: src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoAbbreviatedDayNames.cs (hunk 1 above)
Comment (tarekgh) [inline src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoAbbreviatedDayNames.cs]: thanks @jkotas could you please close the issue [https://github.com/dotnet/corefx/issues/13247](https://github.com/dotnet/corefx/issues/13247) then?

---
In reply to: [86171814](https://github.com/dotnet/corefx/pull/13229#discussion_r86171814) [](ancestors = 86171814)

Comment (tarekgh) [inline src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoAbbreviatedDayNames.cs]: thanks @jkotas could you close https://github.com/dotnet/corefx/issues/13247 if you already fixed this issue?

**Comment by tarekgh** [comment]
should we pick up the new built coreclr which contains @jkotas changes?

**Comment by JeremyKuhne** [comment]
I'll pick up the latest CoreCLR and re-up.

File: src/System.Runtime/ref/System.Runtime.cs (hunk 9 above)
Comment (JeremyKuhne) [inline src/System.Runtime/ref/System.Runtime.cs]: I'm fine with that, but I'll do that in a separate change. @weshaggard Should we keep tests where the CoreCLR code is first mapped in?

File: src/System.Text.Encoding/tests/Fallback/EncoderReplacementFallbackTests.cs (hunk 8 above)
Comment (JeremyKuhne) [inline src/System.Text.Encoding/tests/Fallback/EncoderReplacementFallbackTests.cs]: I'll undo it.

File: src/System.Runtime/tests/System/Text/StringBuilderTests.cs (hunk 7 above)
Comment (JeremyKuhne) [inline src/System.Runtime/tests/System/Text/StringBuilderTests.cs]: I'll update to the latest CoreCLR.

File: src/System.Runtime/ref/System.Runtime.cs (hunk 6 above)
Comment (JeremyKuhne) [inline src/System.Runtime/ref/System.Runtime.cs]: Ok.

File: src/System.Net.Http/tests/UnitTests/System.Net.Http.Unit.Tests.csproj (hunk 5 above)
Comment (JeremyKuhne) [inline src/System.Net.Http/tests/UnitTests/System.Net.Http.Unit.Tests.csproj]: Ok, thanks.

File: src/System.IO.FileSystem/src/System/IO/UnixFileSystem.cs (hunk 4 above)
Comment (JeremyKuhne) [inline src/System.IO.FileSystem/src/System/IO/UnixFileSystem.cs]: I'll try and find a good place for it in another PR.  I might create an InternalConstants class in IO- thoughts?

File: src/System.IO.FileSystem/src/System/IO/ReadLinesIterator.cs [hunk 11]
```diff
@@ -97,7 +97,7 @@ internal static ReadLinesIterator CreateIterator(string path, Encoding encoding)
 
         private static ReadLinesIterator CreateIterator(string path, Encoding encoding, StreamReader reader)
         {
-            return new ReadLinesIterator(path, encoding, reader ?? new StreamReader(FileStream.InternalOpen(path, useAsync: false), encoding));
+            return new ReadLinesIterator(path, encoding, reader ?? new StreamReader(new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read), encoding));
```
Comment (JeremyKuhne) [inline src/System.IO.FileSystem/src/System/IO/ReadLinesIterator.cs]: Good question. I'll look into it.

File: src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoAbbreviatedMonthGenitiveNames.cs (hunk 2 above)
Comment (JeremyKuhne) [inline src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoAbbreviatedMonthGenitiveNames.cs]: Removing.

File: src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoAbbreviatedDayNames.cs (hunk 1 above)
Comment (JeremyKuhne) [inline src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoAbbreviatedDayNames.cs]: Reverting.

Comment (tarekgh) [inline src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoAbbreviatedDayNames.cs]: we need to revert  ActiveIssue(13247, TestPlatforms.AnyUnix) too not just the code. just double checking

---
In reply to: [86198590](https://github.com/dotnet/corefx/pull/13229#discussion_r86198590) [](ancestors = 86198590)

File: src/System.IO.FileSystem/src/System/IO/ReadLinesIterator.cs (hunk 11 above)
Comment (JeremyKuhne) [inline src/System.IO.FileSystem/src/System/IO/ReadLinesIterator.cs]: I'll deal with simplifying this where possible in a subsequent change. There is additional cleanup that can potentially be done with SR/W because FS is moving down.

File: src/System.Runtime/tests/System/Text/StringBuilderTests.cs (hunk 7 above)
Comment (Clockwork-Muse) [inline src/System.Runtime/tests/System/Text/StringBuilderTests.cs]: @stephentoub - yes, it was deliberate.  Anything that was a change was pulled out into a separate commit in the PR.  That's my bad for not noticing it would be a breaking change.  I was expecting the runtime tests to have checked against corefx too...

**Comment by JeremyKuhne** [comment]
Arghhh- updating again for conflicts...

**Comment by JeremyKuhne** [comment]
@dotnet-bot test Innerloop Linux ARM Emulator SoftFP Release Cross Build (setenv_incpath.sh: No such file or directory)

**Comment by JeremyKuhne** [comment]
Update for conflicts, again...

**Comment by JeremyKuhne** [comment]
ValueTuple tests are failing intermittently- investigating

File: src/System.Runtime/ref/System.Runtime.cs (hunk 9 above)
Comment (weshaggard) [inline src/System.Runtime/ref/System.Runtime.cs]: Right now we have been trying to keep them where they were while we still have older platforms that support the older contract that still want to run these tests. They will eventually get baselined away and then merged with where they live now.

File: src/System.IO.FileSystem.Primitives/ref/System.IO.FileSystem.Primitives.cs (hunk 3 above)
Comment (weshaggard) [inline src/System.IO.FileSystem.Primitives/ref/System.IO.FileSystem.Primitives.cs]: Yes we should move them as well and collapse this contract.

**Comment by JeremyKuhne** [comment]
Kicking CoreCLR up to the latest (24702-03)

**Comment by JeremyKuhne** [comment]
@dotnet-bot test Innerloop Windows_NT Release Build and Test (ValueTuple ToString tests failing)

**Comment by joperezr** [comment]
test Innerloop OSX Release Build and Test please
test Innerloop Windows_NT Release Build and Test please (few tests were failing both builds)

**Comment by tarekgh** [comment]
the failures in OSX and WINDOWS_NT innerloop could be legitimate failures. I am not saying these result of @JeremyKuhne changes but it could be the tests are not stable

**Comment by stephentoub** [comment]
> ValueTuple ToString tests failing

If we're repeatedly seeing the ValueTuple tests failing in certain legs here, and we've not seen such failures elsewhere, that would seem to suggest some kind of real issue here, no?

**Comment by tarekgh** [comment]
CC @jcouv for ValueTuple

**Comment by JeremyKuhne** [comment]
> If we're repeatedly seeing the ValueTuple tests failing in certain legs here, and we've not seen such failures elsewhere, that would seem to suggest some kind of real issue here, no?

I see a ton of failures in Jenkins. More than seems plausible for just my PR, but it's hard to be definitive about it as Jenkins is so incredibly slow to drill down into (several minutes to hours). I can't find anything specific in CoreCLR that would cause this sort of behavior, but some of the changes are beyond my understanding. If only I could reproduce the error locally...

@jkotas Do any of the changes lately to CLR look like they might be involved here? This doesn't seem to be happening for any other tests...

Here is one that is failing:

Assert.Equal() Failure\r\n ↓ (pos 1)\r\nExpected: (1, 2, 3)\r\nActual: (0, 0, 0)\r\n ↑ (pos 1)

``` C#
        [Fact]
        public static void ConvertToRef3()
        {
            var refTuple = Tuple.Create(-1, -1, -1);
            var valueTuple = ValueTuple.Create(1, 2, 3);

            refTuple = valueTuple.ToTuple();
            Assert.Equal("(1, 2, 3)", refTuple.ToString());
        }
```

**Comment by stephentoub** [comment]
@dotnet-bot Test Innerloop Windows 10 Release Build and Test please
@dotnet-bot Test Innerloop Windows 7 Release Build and Test please

**Comment by stephentoub** [comment]
@dotnet-bot help

**Comment by stephentoub** [comment]
@dotnet-bot test outerloop Windows 7 Debug
@dotnet-bot test outerloop Windows 7 Release
@dotnet-bot test outerloop Windows_NT Debug
@dotnet-bot test outerloop Windows_NT Release
@dotnet-bot test outerloop Windows 10 Debug
@dotnet-bot test outerloop Windows 10 Release

**Comment by JeremyKuhne** [comment]
@jkotas @stephentoub Release is definitely busted. I'm going to try and track down which CoreCLR change did it.

**Comment by jkotas** [comment]
This looks like a JIT bug to me. https://github.com/dotnet/coreclr/pull/7677 struct optimizations is the most likely culprit.

cc @dotnet/jit-contrib @CarolEidt

**Comment by JeremyKuhne** [comment]
@jkotas @CarolEidt dotnet/coreclr#7677 does look to be the culprit. After https://github.com/dotnet/versions/commit/55b66c64609e99fa1275d4a1ee4d4df96bf516a8#diff-d7af4f169c02eb8bbc36045489048209 the tests start failing.

0e12effd3cb1b542b73855cc9b58b68a3bc3ac76 was the last good CoreCLR version.

**Comment by JeremyKuhne** [comment]
Here are the repro steps in CoreFX:

1. Update `dependencies.props` `<CoreClrCurrentRef>` to the version you want -- such as [55b66c64609e99fa1275d4a1ee4d4df96bf516a8](dotnet/versions@55b66c6#diff-d7af4f169c02eb8bbc36045489048209)

2. Run the following commands from the root of CoreFX

```
rd bin /s /q
build-managed.cmd -- /t:UpdateDependencies
del src\System.ValueTuple\src\project.lock.json
sync.cmd
msbuild src\System.ValueTuple\src\System.ValueTuple.csproj
msbuild src\System.ValueTuple\tests\System.ValueTuple.Tests.builds /t:RebuildAndTest /p:Configuration=Release
```

**Comment by JeremyKuhne** [comment]
Moved this change to the build prior to the regression (apparently dotnet/coreclr#7677). We're blocked from going any further forward until we deal with the regression.

Note that I did validate the most recent drop of CoreCLR has the same failure.

**Comment by jkotas** [comment]
Discussed with @RussKeldorph .... we are most likely going to rollback the offending change.

**Comment by RussKeldorph** [comment]
@jkotas Go ahead and back out.  @pgavlin can help if there are conflicts stacked on top of it.

**Comment by JeremyKuhne** [comment]
@jkotas I'll update this PR again once the change is pulled. I forgot I'm dependent on the other fix around exception strings.

**Comment by jkotas** [comment]
Backout PR: https://github.com/dotnet/coreclr/pull/7982

**Comment by JeremyKuhne** [comment]
Ok, _hopefully_ we'll be good now. Had to deal with ARM64 builds being disabled before we could get a full set of packages. I've updated the PR to 24704-03.

**Comment by stephentoub** [comment]
Thanks for working through the issues, @JeremyKuhne.
